notifier.cpp

```diff
@@ -100,12 +104,21 @@ int main(int argc, char** argv) {
         // Note that if this call fails, then we need to free the pubMsg object here since it won't be passed to the ack handler.
         // if (s != NATS_OK)
         //     free(pubMsg);
+
+        done_mtx.lock();
+        if (done) {
+            done_mtx.unlock();
+            break;
+        }
+        done_mtx.unlock();
     }
 
+    done_mtx.lock();
```

Comment:
btw, there is some build-in locking in nats-streaming: https://github.com/nats-io/nats.c/blob/f0d2bb4626c66d5e36defbc42e7cc6f402b83b97/src/stan/conn.h#L28-L29
looks like it's preferred way if you want to lock connection itself. but not sure